File: java/src/test/java/problems1_100/FindTheIndexOfTheFirstOccurrenceInAStringTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class FindTheIndexOfTheFirstOccurrenceInAStringTest extends SampledTest {
    @Test
    public void test() {
        for (FindTheIndexOfTheFirstOccurrenceInAString solution
            : FindTheIndexOfTheFirstOccurrenceInAString.values()) {
            for (FindTheIndexOfTheFirstOccurrenceInAStringSample sample
                : getSamples(FindTheIndexOfTheFirstOccurrenceInAStringSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.strStr(
                            sample.input.haystack,
                            sample.input.needle
                        )
                    ).isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/FindTheIndexOfTheFirstOccurrenceInAStringTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class FindTheIndexOfTheFirstOccurrenceInAStringTest : SampledTest() {
    @Test
    fun test() =
        FindTheIndexOfTheFirstOccurrenceInAString.entries.forEach { solution ->
            getSamples(Array<FindTheIndexOfTheFirstOccurrenceInAStringSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.strStr(input.haystack, input.needle))
                        .isEqualTo(output)
                }
        }
}
